tests/test_normalizedCalibrationFlux.py

```diff
@@ -0,0 +1,270 @@
+                self.assertLess(ratio, 1.0)
+
+            # The subset that was used should always have a 1.0 ratio, though
+            # this may be not quite zero because of the trend in the ratio
+            # vs flux even at the bright end.
+            used = catalog["apcorr_base_CompensatedTophatFlux_12_used"]
+            ratio_used = np.median(
+                catalog["slot_CalibFlux_instFlux"][used]/catalog[self.ap_name + "_instFlux"][used]
+            )
+            self.assertFloatsAlmostEqual(ratio_used, 1.0, rtol=1e-10)
+
+    def testNormalizedCalibrationFluxTooFew(self):
+        np.random.seed(12345)
+        norm_task = self._make_task()
+        catalog = self._make_catalog(norm_task.schema)
+
+        flags = np.ones(len(catalog), dtype=bool)
+        flags[0] = False
+        catalog[self.cg_name + "_flag"] = flags
+
+        ap_corr_map = norm_task.run(catalog=catalog, exposure=self.exposure).ap_corr_map
+
+        self.assertIn("base_CompensatedTophatFlux_12_instFlux", ap_corr_map)
+        self.assertIn("base_CompensatedTophatFlux_12_instFluxErr", ap_corr_map)
+
+        # The full set should have a 1.0 ratio when the
+        # aperture flux offset is 0.0
+        ratio = np.mean(catalog["slot_CalibFlux_instFlux"]/catalog[self.ap_name + "_instFlux"])
+        self.assertFloatsAlmostEqual(ratio, 1.0, rtol=1e-10)
+
+        self.assertTrue(np.all(~flags == catalog["apcorr_base_CompensatedTophatFlux_12_used"]))
+
+    def testNormalizedCalibrationFluxApplyOnly(self):
+        # Run the regular task in default mode first.
+        np.random.seed(12345)
+        norm_task = self._make_task()
+        catalog_run1 = self._make_catalog(norm_task.schema)
+        exposure_run1 = self.exposure.clone()
+
+        ap_corr_map = norm_task.run(catalog=catalog_run1, exposure=exposure_run1).ap_corr_map
+
+        exposure_run1.info.setApCorrMap(ap_corr_map)
+
+        # Rerun the task; we need to make sure we have the same input so re-seed.
+        np.random.seed(12345)
+        norm_task2 = self._make_task(apply_only=True)
+        catalog_run2 = self._make_catalog(norm_task.schema)
+
+        ap_corr_map2 = norm_task2.run(catalog=catalog_run2, exposure=exposure_run1).ap_corr_map
+
+        # Check that the ap_corr_map and ap_corr_map2 are the same.
+        self.assertEqual(set(ap_corr_map2.keys()), set(ap_corr_map.keys()))
+        for key in ap_corr_map.keys():
+            self.assertEqual(ap_corr_map2[key], ap_corr_map[key])
+
+        # Check that the slot is set correctly
+        self.assertEqual(
+            catalog_run2.schema.getAliasMap().get("slot_CalibFlux"),
+            norm_task2.config.normalized_calibflux_name,
+        )
+
+        # Check that the final normalized catalog values are the same.
+        self.assertFloatsAlmostEqual(
+            catalog_run2["slot_CalibFlux_instFlux"],
+            catalog_run1["slot_CalibFlux_instFlux"],
+        )
+
+    def testNormalizedCalibrationFluxApplyOnlyFail(self):
```

Comment:
Another nitpick but is it conventional to include the name of the test case class in the test member methods?

Reply:
Are these too wordy?  I tend to get verbose with my test names because it's much better than being too terse, but if you think I should shorten them I can.